src/Api/RatingService.php

```diff
@@ -82,15 +68,15 @@ public function getRates(ShipmentInterface $shipment) {
 
     try {
       $request = new Rating($config);
-      $response = $request->getRates($origin_postal_code, $postal_code, $weight);
+      $response = $request->getRates($origin_postal_code, $postal_code, $weight, $options);
     }
     catch (ClientException $exception) {
       $message = sprintf(
         'An error has been returned by the Canada Post when fetching the shipping rates. The error was: "%s"',
         json_encode($exception->getResponseBody())
       );
       $this->logger->error($message);
-      return;
+      return [];
```

Comment:
**CHANGE REQUEST**

It's best that you return either `NULL` or `FALSE` so that the caller can handle errors if needed. Could you check if `commerce_shipping` will do anything if the shipping method plugin does not return any rates (empty array)?

I think it will simply not present any rates for this shipping method, and that might not be the best. The best would be to return FALSE from the rate service and catch that in the shipping method. Then display an error message like "Some of the shipping options could not be calculated at this time, please try again. If the error persists don't hesitate to let us know.". And then return an empty array that `commerce_shipping` would expect.

So, I'd suggest that you revert to return NULL (unless this causes an error) and open another issue on DO for handling such error.